modules/canbus/vehicle/ch/ch_controller.cc

```diff
@@ -0,0 +1,575 @@
+  const int32_t flag =
+      CHECK_RESPONSE_STEER_UNIT_FLAG | CHECK_RESPONSE_SPEED_UNIT_FLAG;
+  if (!CheckResponse(flag, true)) {
+    AERROR << "Failed to switch to COMPLETE_AUTO_DRIVE mode.";
+    Emergency();
+    set_chassis_error_code(Chassis::CHASSIS_ERROR);
+    return ErrorCode::CANBUS_ERROR;
+  } else {
+    set_driving_mode(Chassis::COMPLETE_AUTO_DRIVE);
+    AINFO << "Switch to COMPLETE_AUTO_DRIVE mode ok.";
+    return ErrorCode::OK;
+  }
+}
+
+ErrorCode ChController::DisableAutoMode() {
+  ResetProtocol();
+  can_sender_->Update();
+  set_driving_mode(Chassis::COMPLETE_MANUAL);
+  set_chassis_error_code(Chassis::NO_ERROR);
+  AINFO << "Switch to COMPLETE_MANUAL OK!";
+  return ErrorCode::OK;
+}
+
+ErrorCode ChController::EnableSteeringOnlyMode() {
+  AFATAL << "SteeringOnlyMode Not supported!";
+  return ErrorCode::OK;
+}
+
+ErrorCode ChController::EnableSpeedOnlyMode() {
+  AFATAL << "SpeedOnlyMode Not supported!";
+  return ErrorCode::OK;
+}
+
+// NEUTRAL, REVERSE, DRIVE
+void ChController::Gear(Chassis::GearPosition gear_position) {
+  if (!(driving_mode() == Chassis::COMPLETE_AUTO_DRIVE ||
+        driving_mode() == Chassis::AUTO_SPEED_ONLY)) {
+    AINFO << "this drive mode no need to set gear.";
+    return;
+  }
+
+  // ADD YOUR OWN CAR CHASSIS OPERATION
+  switch (gear_position) {
+    case Chassis::GEAR_NEUTRAL: {
+      gear_command_114_->set_gear_cmd(Gear_command_114::GEAR_CMD_NEUTRAL);
+      break;
+    }
+    case Chassis::GEAR_REVERSE: {
+      gear_command_114_->set_gear_cmd(Gear_command_114::GEAR_CMD_REVERSE);
+      break;
+    }
+    case Chassis::GEAR_DRIVE: {
+      gear_command_114_->set_gear_cmd(Gear_command_114::GEAR_CMD_DRIVE);
+      break;
+    }
+    case Chassis::GEAR_PARKING: {
+      gear_command_114_->set_gear_cmd(Gear_command_114::GEAR_CMD_PARK);
+      break;
+    }
+    case Chassis::GEAR_INVALID: {
+      AERROR << "Gear command is invalid!";
+      gear_command_114_->set_gear_cmd(Gear_command_114::GEAR_CMD_NEUTRAL);
+      break;
+    }
+    default: {
+      gear_command_114_->set_gear_cmd(Gear_command_114::GEAR_CMD_NEUTRAL);
+      break;
+    }
+  }
+}
+
+// brake with new acceleration
+// acceleration:0.00~99.99, unit:
+// acceleration:0.0 ~ 7.0, unit:m/s^2
+// acceleration_spd:60 ~ 100, suggest: 90
+// -> pedal
+void ChController::Brake(double pedal) {
+  // double real_value = params_.max_acc() * acceleration / 100;
+  // Update brake value based on mode
+  if (!(driving_mode() == Chassis::COMPLETE_AUTO_DRIVE ||
+        driving_mode() == Chassis::AUTO_SPEED_ONLY)) {
+    AINFO << "The current drive mode does not need to set acceleration.";
+    return;
+  }
+  // ADD YOUR OWN CAR CHASSIS OPERATION
+  brake_command_111_->set_brake_pedal_cmd(static_cast<int>(pedal));
+}
+
+// drive with old acceleration
+// gas:0.00~99.99 unit:
+void ChController::Throttle(double pedal) {
+  if (!(driving_mode() == Chassis::COMPLETE_AUTO_DRIVE ||
+        driving_mode() == Chassis::AUTO_SPEED_ONLY)) {
```

Comment:
`driving_mode != COMPLETE_AUTO_DRIVE && driving_mode != AUTO_SPEED_ONLY`

Reply:
I have already modified it, please review it again. Think you!